src/jaxoplanet/orbits/keplerian.py

```diff
@@ -641,71 +604,90 @@ def _get_position_and_velocity(
             v1, v2, include_inclination=semiamplitude is None
         )
 
-        return (x, y, z), (vx, vy, vz)
+        # In the case a semiamplitude was passed without units, we strip the
+        # units of the output
+        if semiamplitude is not None and not (
+            hasattr(semiamplitude, "_magnitude") and hasattr(semiamplitude, "_units")
+        ):
+            vx = vx.magnitude
+            vy = vy.magnitude
+            vz = vz.magnitude
 
+        return (x, y, z), (vx, vy, vz)
 
-class KeplerianOrbit(NamedTuple):
-    bodies: KeplerianBody
 
-    def __len__(self) -> int:
-        return len(self.bodies.period)
+class System(eqx.Module):
+    central: Central
+    bodies: tuple[Body, ...]
+    _body_stack: Optional[Body]
+
+    def __init__(
+        self, central: Optional[Central] = None, *, bodies: tuple[Body, ...] = ()
+    ):
+        self.central = Central() if central is None else central
+        self.bodies = bodies
+
+        # If all the bodies have matching Pytree structure then we save a
+        # stacked version that we can use for vmaps below. This allows for more
+        # efficient evaluations in the case of multiple bodies.
+        self._body_stack = None
+        if len(bodies):
+            spec = list(map(jax.tree_util.tree_structure, bodies))
+            if spec.count(spec[0]) == len(spec):
+                self._body_stack = jax.tree_util.tree_map(
+                    lambda *x: jnp.stack(x, axis=0), *bodies
+                )
```

Comment:
@lgrcia — You might find this interesting... I've managed to work out a way to have the interface that you suggested (which I like very much!), while also getting the good performance of being able to `vmap` over bodies. There are some interesting edge cases where the `vmap` wouldn't work (for example, if some bodies have eccentricity defined, but others don't), and we transparently fall back on a potentially less performant, but general method.